File: backend/src/main/scala/order/api/ReadyOrderApi.scala
package order.api

import domain.shared.given

import cats.effect.IO
import domain.shared.{DeliveryAppState, OrderId, UserRole}
import order.app.readyOrder
import org.http4s.HttpRoutes
import org.http4s.circe.CirceEntityEncoder.*
import org.http4s.dsl.io.*
import shared.api.routing.*
import shared.app.*

val readyOrderApi: FixedMethodApi1[OrderId, DeliveryAppState] =
  jsonPostApi1[OrderId, Unit, DeliveryAppState](
    List(routeSegment("api"), routeSegment("delivery"), routeSegment("orders")),
    List(routeSegment("ready")),
  )

File: frontend/src/apis/order/ReadyOrderApi.ts
import type { DeliveryAppState, OrderId } from '@/objects/core/SharedObjects'
import { postNormalizedDeliveryStateWithoutBody } from '@/system/api/DeliveryStateClient'
import { buildApiPath1, defineJsonPostApi1, routeSegment } from '@/system/api/TypedApiDefinitions'

export const readyOrderApiDefinition = defineJsonPostApi1<OrderId, void, DeliveryAppState>(
  [routeSegment('api'), routeSegment('delivery'), routeSegment('orders')],
  [routeSegment('ready')],
)

export function readyOrder(orderId: OrderId) {
  return postNormalizedDeliveryStateWithoutBody(buildApiPath1(readyOrderApiDefinition, orderId))
}
